### PR TITLE
feat: implement automatic trash cleanup

### LINKED ISSUE #1931: Time-based delete from archive not working
**Describe the bug**
According to #1625 a deleted goal should be 'deleted forever' after spending 7 days in the archive. This is not working.

**To Reproduce**
Steps to reproduce: Delete, change system date, open app, see it's not deleted.

**Expected behavior**
Delete any goals/budgets in trash after 7 days.

**Screenshots**
None.

**Additional context**
None.

### PR DESCRIPTION
**Pull request by vinaybadgujar102**
resolves #1931

### CODE REVIEW

**Review by tijlleenders — DISMISSED** [review]
Cool, I will check this in dev after we merge.

@Tushar-4781 Code OK?

File: src/api/TrashAPI/index.ts [hunk 1]
```diff
@@ -87,3 +88,35 @@ export const getParticipantsOfDeletedGoal = async (id: string) => {
...
+
+export const cleanupTrash = async () => {
+  const sevenDaysAgo = new Date();
+  sevenDaysAgo.setDate(sevenDaysAgo.getDate() - TRASH_RETENTION_DAYS);
+
+  const oldTrashItems = await db.goalTrashCollection.where("deletedAt").below(sevenDaysAgo.toISOString()).toArray();
+
+  await Promise.all(oldTrashItems.map((item) => db.goalTrashCollection.delete(item.id)));
+
+  console.log(`Deleted ${oldTrashItems.length} items from trash older than ${TRASH_RETENTION_DAYS} days.`);
```
Comment (Tushar-4781) [inline src/api/TrashAPI/index.ts:100]: we can remove this log

File: src/api/TrashAPI/index.ts [hunk 2]
```diff
@@ -87,3 +88,35 @@ export const getParticipantsOfDeletedGoal = async (id: string) => {
...
 };
+
+export const cleanupTrash = async () => {
+  const sevenDaysAgo = new Date();
+  sevenDaysAgo.setDate(sevenDaysAgo.getDate() - TRASH_RETENTION_DAYS);
+
+  const oldTrashItems = await db.goalTrashCollection.where("deletedAt").below(sevenDaysAgo.toISOString()).toArray();
+
+  await Promise.all(oldTrashItems.map((item) => db.goalTrashCollection.delete(item.id)));
+
```
Comment (Tushar-4781) [inline src/api/TrashAPI/index.ts:99]: no need to leave line breaks

File: src/hooks/useApp.tsx [hunk 3]
```diff
@@ -143,6 +144,10 @@ function useApp() {
       });
   }, []);
 
+  useEffect(() => {
+    initializeTrashManagement();
```
Comment (Tushar-4781) [inline src/hooks/useApp.tsx:148]: u can directly call checkAndCleanupTrash

**Review by Tushar-4781 — CHANGES_REQUESTED** [review]

**Comment by vinaybadgujar102** [comment]
@Tushar-4781 made the requested changes in the latest commit

**Review by Tushar-4781 — APPROVED** [review]

**Pull request merged**